docs/source/distributed.rst

```diff
@@ -629,7 +629,7 @@ The following error message is produced on rank 0, allowing the user to determin
 ``TORCH_DISTRIBUTED_DEBUG``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Next, the environment variable ``TORCH_DISTRIBUTED_DEBUG``  can be used to trigger additional useful logging and collective synchronization checks to ensure all ranks
+With ``TORCH_CPP_LOG_LEVEL=INFO``, the environment variable ``TORCH_DISTRIBUTED_DEBUG``  can be used to trigger additional useful logging and collective synchronization checks to ensure all ranks
```

Comment:
This statement is true for "additional useful logging", but `TORCH_DISTRIBUTED_DEBUG` can be used independently from `TORCH_CPP_LOG_LEVEL` to enable collective synchronization checks.

A bit background on this might be helpful: In an ideal world we wouldn't intermix the two environment variables, but unfortunately the number of logging levels we have in our C++ code base is not sufficient enough for all of our use cases. We have had several customers who wanted to have very granular log output beyond `INFO` for troubleshooting their distributed training jobs. So we took the existing `TORCH_DISTRIBUTED_DEBUG` variable that was already used for collective synchronization checks and leveraged it to define additional verbose log levels when used in combination with `TORCH_CPP_LOG_LEVEL` as described [here](https://pytorch.org/docs/master/distributed.html#logging).

The existing behavior of `TORCH_DISTRIBUTED_DEBUG` has not been changed by this extra use though. So for running collective synchronization checks, the existence or value of `TORCH_CPP_LOG_LEVEL` should be irrelevant.

Reply:
`TORCH_DISTRIBUTED_DEBUG` can indeed be use independently as @cbalioglu mentioned, although `TORCH_CPP_LOGLEVEL=INFO` is practically needed for the full feature set of torch_distributed_debug. For example, torch_distributed_debug does `INFO` logging for things such as DDP init, so `TORCH_CPP_LOGLEVEL=INFO` should be set.

In practice, https://github.com/pytorch/pytorch/pull/71746 enabled `TORCH_DISTRIBUTED_DEBUG` to be much more practically useful for OSS users as they no longer have to build from source!